agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -376,3 +456,168 @@ func (cs *CredentialSpecResource) MarshallPlatformSpecificFields(credentialSpecR
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("file://%s", filepath.Join(dir, copiedFileName)), nil
+}
+
+func setTaskExecutionCredentialsRegKeys(taskCredentials credentials.IAMRoleCredentials, taskArn string) error {
+	if taskCredentials == (credentials.IAMRoleCredentials{}) {
+		err := errors.New("Unable to find execution role credentials while setting registry key")
+		return err
+	}
+
+	seelog.Errorf("SetTaskExecutionCredentialsRegKey")
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, servicesRegistryKeyRoot, registry.WRITE)
+	if err != nil {
+		seelog.Errorf("Error opening %s key: %s", servicesRegistryKeyRoot, err)
+		return err
+	}
+	defer k.Close()
+
+	k2, _, err := registry.CreateKey(k, taskCredentialsRegistryKey, registry.WRITE)
+	if err != nil {
+		seelog.Errorf("Error creating %s key: %s", taskCredentialsRegistryKey, err)
+		return err
+	}
+	defer k2.Close()
+
+	k3, _, err := registry.CreateKey(k2, taskArn, registry.WRITE)
+	if err != nil {
+		seelog.Errorf("Error creating %s key: %s", taskArn, err)
+		return err
+	}
+	defer k3.Close()
+
+	err = k3.SetStringValue("AKID", taskCredentials.AccessKeyID)
+	if err != nil {
+		seelog.Errorf("Error creating AKID child value for task %s:%s", taskArn, err)
+		return err
+	}
+	err = k3.SetStringValue("SKID", taskCredentials.SecretAccessKey)
+	if err != nil {
+		seelog.Errorf("Error creating SKID child value for task %s:%s", taskArn, err)
+		return err
+	}
+	err = k3.SetStringValue("SESSIONTOKEN", taskCredentials.SessionToken)
+	if err != nil {
+		seelog.Errorf("Error creating SESSIONTOKEN child value for task %s:%s", taskArn, err)
+		return err
+	}
+
+	return nil
+}
+
+func handleNonFileDomainlessGMSACredSpec(originalCredSpec, localCredSpecFilePath, taskARN string) error {
+	// Exit early for non domainless gMSA cred specs
+	if !strings.HasPrefix(originalCredSpec, "credentialspecdomainless:") {
+		return nil
+	}
+
+	err := readWriteDomainlessCredentialSpec(localCredSpecFilePath, localCredSpecFilePath, taskARN)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func readWriteDomainlessCredentialSpec(filePath, outFilePath, taskARN string) error {
+	credSpec, err := readCredentialSpec(filePath)
+	if err != nil {
+		return err
+	}
+	err = writeCredentialSpec(credSpec, outFilePath, taskARN)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func readCredentialSpec(filePath string) (map[string]interface{}, error) {
+	byteResult, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+	var credSpec map[string]interface{}
+	err = json.Unmarshal(byteResult, &credSpec)
+	if err != nil {
+		return nil, err
+	}
+	return credSpec, nil
+}
+
+func writeCredentialSpec(credSpec map[string]interface{}, outFilePath string, taskARN string) error {
+	if activeDirectoryConfigUntyped, ok := credSpec["ActiveDirectoryConfig"]; ok {
+		activeDirectoryConfig := activeDirectoryConfigUntyped.(map[string]interface{})
```

Comment:
This line will panic if `activeDirectoryConfigUntyped` is not of type `map[string]interface{}`. Can we add a check please.

Reply:
added in next PR